networkx/algorithms/covering.py

```diff
@@ -0,0 +1,63 @@
+"""
+An edge cover of a graph is a set of edges such that every vertex of
+the graph is incident to at least one edge of the set.
+A minimum edge cover is an edge covering of smallest possible size.
+
+`Edge Cover <https://en.wikipedia.org/wiki/Edge_cover>`_
+"""
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def min_edge_cover(G):
+    """Returns a set of edges which constitute the minimum edge cover of the graph.
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+
+    Returns
+    -------
+    min_cover : set
+        The covering is returned as a set, ``min_cover``.
+        It contains all the edges of minimum edge cover
+        in form of tuples.
+
+    Notes
+    -----
+    Worst Case runtime of ``nx.max_weight_matching(G, maxcardinality=True)``
+    is `O(number_of_nodes^{3})`
+    And the greedy extension works in `O(n\log{}n)`
+    So worst case run time of the function is `O(number_of_nodes^{3})`.
+    """
+    maximum_matching = {}
+    is_cover_exist = True
+    for node in G.nodes():
+        if not G[node]:
+            is_cover_exist = False
+            break
+    if not is_cover_exist:
+        # min_cover does not exist
+        # raise an exception
+        raise nx.NetworkXException("No Edge Cover Exist")
+    maximum_matching = nx.max_weight_matching(G, maxcardinality=True)
+    min_cover = set()
+    # ``min_cover`` is superset of ``maximum_matching``
+    for node in maximum_matching:
+        min_cover.add((node, maximum_matching[node]))
+
+    # iterate for uncovered nodes
+    for node in set(G.nodes()) - set(x[0] for x in min_cover):
```

Comment:
How about this instead? I think it's more readable:

    from networkx.utils import arbitrary_element
    ...
        uncovered_nodes = set(G) - {v for u, v in min_cover}
        for u in uncovered_nodes:
            # Since `v` is uncovered, each edge incident to `v` will join it
            # with a covered node (otherwise, if there were an edge joining
            # uncovered nodes `u` and `v`, the maximum matching algorithm
            # would have found it), so we can choose an arbitrary edge
            # incident to `v`. (This applies only in a simple graph, not a
            # multigraph.)
            #
            # Since the graph is guaranteed to be undirected, we need only
            # add a single orientation of the edge.
            v = arbitrary_element(G[v])
            min_cover.add((u, v))